src/Npgsql.EntityFrameworkCore.PostgreSQL/Migrations/Internal/NpgsqlHistoryRepository.cs

```diff
@@ -90,17 +90,27 @@ public override string GetCreateIfNotExistsScript()
 
         public override string GetBeginIfNotExistsScript(string migrationId)
         {
-            throw new NotSupportedException("Generating idempotent scripts for migration is not currently supported by Npgsql");
+            return @"
+DO $$
+BEGIN
+    IF NOT EXISTS(SELECT 1 FROM public.""__EFMigrationsHistory"" where ""MigrationId"" = '{migrationId}') THEN"
```

Comment:
Please don't hardcode the schema or table name, use `TableSchema` and `TableName` instead (with string interpolation)